src/Helpers/moduleStartStopCatch.ts

```diff
@@ -0,0 +1,7 @@
+import * as Pino from 'pino'
+
+export const shutdownModule = (server: any) =>
```

Comment:
We can avoid this `any` if we declare an interface for `start()` and `stop()` and have the `interface BlockchainWriter` extend from it.

Maybe we could call it [Lifecycle](https://docs.spring.io/spring/docs/3.2.x/javadoc-api/org/springframework/context/Lifecycle.html) or StartStoppable?

```ts
interface Lifecycle {
  readonly start: () => Promise<void>
  readonly stop: () => Promise<void>
}
```

Reply:
Or it could just take a stop callback instead of the entire object, I can't think of a reason why this function would need to know about the entire object.

Also maybe rename this function, shutdownModule sounds like your commanding it to shut down, but it's really an event handler so maybe something like `onShutdownEvent` or `handleShutdownEvent`. There may be a better name those are just a couple off the top of my head.

Reply:
I'll play with it. I had an interface defined, but it was complaining at the time. I took it out, but I think it was an unrelated issue, and I didn't get back to it.